src/verify.ts

```diff
@@ -0,0 +1,231 @@
+import { join } from "node:path";
+
+import {
+  SubManifest,
+  SubManifestContentEntry,
+  SuperManifest,
+} from "./manifest.js";
+
+export interface VerificationFile {
+  hash: string;
+  cid: string;
+  byteLength: number;
+}
+
+export interface VerificationSplitFile {
+  name: string;
+  hash: string;
+  byteLength: number;
+  parts: VerificationFilePart[];
+}
+
+export interface VerificationFilePart {
+  name: string;
+  byteLength: number;
+  cid: string;
+  rootCID: string;
+  originalFileName: string;
+  originalFileHash: string;
+  originalFileByteLength: number;
+}
+
+export type VerificationDirectory = null;
+
+export type VerificationContentEntry =
+  | VerificationFile
+  | VerificationSplitFile
+  | VerificationDirectory;
+
+function manifestKeyCheck(
+  sup: SuperManifest,
+  sub: SubManifest,
+  key: keyof SubManifest & keyof SuperManifest
+) {
+  if (sub[key] !== sup[key]) {
+    throw new Error(
+      `Verification failed during manifestKeyCheck, expected '${JSON.stringify(sup[key])}', got '${JSON.stringify(sub[key])}'`
+    );
+  }
+}
+
+export class PieceVerifier {
+  #carFile: string;
+  #files: Map<string, VerificationFile>;
+  #dirs: Map<string, VerificationDirectory>;
+  #subManifest: SubManifest | undefined;
+
+  constructor(carFile: string) {
+    this.#carFile = carFile;
+    this.#files = new Map<string, VerificationFile>();
+    this.#dirs = new Map<string, VerificationDirectory>();
+  }
+
+  getCarFile(): string {
+    return this.#carFile;
+  }
+
+  getSubManifest(): SubManifest {
+    if (!this.#subManifest) {
+      throw new Error("getManifest() called before verify");
+    }
+    return this.#subManifest;
+  }
+
+  addFile(path: string, info: VerificationFile) {
+    this.#files.set(path, info);
+  }
+
+  addDirectory(path: string) {
+    this.#dirs.set(path, null);
+  }
+
+  verify(subManifest: SubManifest, rootCID: string): VerificationFilePart[] {
+    const fileParts: VerificationFilePart[] = [];
+    this.#subManifest = subManifest;
+
+    if (!subManifest.contents) {
+      return [];
+    }
+
+    console.log("files", this.#files);
+    console.log("dirs", this.#dirs);
```

Comment:
Is this just temporary debug? Otherwsie I think you added `--verbose` to control this behaviour didn't you?